### PR TITLE
Merge master to release 2.5.0

### PR DESCRIPTION
**Pull request by ericclappier-eaton**

### CODE REVIEW

File: pack/map.h [hunk 1]
```diff
@@ -105,6 +106,7 @@ class Map : public IObjectMap
     T&                 append(const std::string& key);
     void               append(const std::string& key, const T& val);
     const Attribute&   get(const std::string& key) const override;
+    void               set(const std::string& key, T& val);
```
Comment (etnKlendathu) [inline pack/map.h:109]: Why val is not const?

File: pack/map.h [hunk 2]
```diff
@@ -147,10 +149,12 @@ class ValueMap : public IValueMap
     void           setValue(const MapType& val);
     bool           contains(const std::string& key) const;
     const CppType& operator[](const std::string& key) const;
+    CppType&       operator[](const std::string& key);
     int            size() const;
     ValueMap&      operator=(const ValueMap& other);
     ValueMap&      operator=(const MapType& val);
     void           append(const std::string& key, const CppType& val);
+    void           set(const std::string& key, CppType& val);
```
Comment (etnKlendathu) [inline pack/map.h:157]: Why val is not const?

File: pack/map.h [hunk 3]
```diff
@@ -213,7 +217,21 @@ int Map<T>::size() const
 template <typename T>
 const T& Map<T>::operator[](const std::string& key) const
 {
-    auto found = std::find_if(m_value.begin(), m_value.end(), [&](const auto& pair) {
+    const auto found = std::find_if(m_value.begin(), m_value.end(), [&](const auto& pair) {
+        return pair.first == key;
+    });
```
Comment (etnKlendathu) [inline pack/map.h:222]: You will search by key? Use https://en.cppreference.com/w/cpp/container/map/find or https://en.cppreference.com/w/cpp/container/map/count

File: pack/map.h [hunk 4]
```diff
@@ -213,7 +217,21 @@ int Map<T>::size() const
...
+
+    if (found != m_value.end()) {
+        return found->second;
+    }
+
+    throw std::out_of_range("Key " + key + " was not found");
```
Comment (etnKlendathu) [inline pack/map.h:228]: fmt::format is better than concatination

File: pack/map.h [hunk 5]
```diff
@@ -339,6 +357,20 @@ const Attribute& Map<T>::get(const std::string& key) const
...
+void Map<T>::set(const std::string& key, T& val)
+{
+    auto found = std::find_if(m_value.begin(), m_value.end(), [&](const auto& pair) {
+        return pair.first == key;
+    });
+
+    if (found != m_value.end()) {
+        found->second = val;
+    } else {
+        throw std::out_of_range("Key " + key + " was not found");
```
Comment (etnKlendathu) [inline pack/map.h:370]: I don't get it. If now such key in map, throw exception? Why? We need to insert it to the map, right?

File: pack/map.h [hunk 6]
```diff
@@ -487,6 +530,17 @@ void ValueMap<ValType>::append(const std::string& key, const CppType& val)
...
 }
 
+template <Type ValType>
+void ValueMap<ValType>::set(const std::string& key, CppType& val)
+{
+    auto found = m_value.find(key);
+    if (found != m_value.end()) {
+        found->second = val;
+    } else {
+        throw std::out_of_range("Key " + key + " was not found");
```
Comment (etnKlendathu) [inline pack/map.h:540]: The same comment as for ObjectMap

File: pack/map.h [hunk 7]
```diff
@@ -339,6 +357,20 @@ const Attribute& Map<T>::get(const std::string& key) const
     throw std::out_of_range("Key " + key + " was not found");
```
Comment (etnKlendathu) [inline pack/map.h:357]: I'm not sure if we need to check and throw exception in this case, because std::map - Returns a reference to the value that is mapped to a key equivalent to key, performing an insertion if such key does not already exist.

File: tests/map.cpp [hunk 8]
```diff
@@ -177,7 +177,7 @@ TEST_CASE("Simple map serialization/deserialization")
     origin.ints.append("key1", 12);
     origin.ints.append("key2", 13);
 
-    auto check = [](const TestMap& item) {
+    auto check = [](TestMap& item) {
```
Comment (etnKlendathu) [inline tests/map.cpp:180]: Why did you remove const? [] operator has const variant

File: tests/map.cpp [hunk 9]
```diff
@@ -241,14 +241,20 @@ struct MapObj: public pack::Node
...
+    CHECK(origin["key1"].value == "Some string modified");
+    origin.set("key1", sav);
+    CHECK(origin["key1"].value == "Some string");
 
     MapObj ins;
     ins.value = "Some other value";
     origin.append("key2", ins);
 
-    auto check = [](const pack::Map<MapObj>& item) {
+    auto check = [](pack::Map<MapObj>& item) {
```
Comment (etnKlendathu) [inline tests/map.cpp:257]: Why did you remove const?

File: tests/map.cpp [hunk 10]
```diff
@@ -263,16 +269,58 @@ TEST_CASE("Object map serialization/deserialization")
...
+
+    auto sav       = origin["key1"];
+    origin["key1"] = "Some string modified";
+    CHECK(origin["key1"] == "Some string modified");
+    origin.set("key1", sav);
+    CHECK(origin["key1"] == "Some string");
+
+    origin.append("key2", "Some other value");
+
+    auto check = [](pack::StringMap& item) {
```
Comment (etnKlendathu) [inline tests/map.cpp:299]: const

**Review by perrettecl — CHANGES_REQUESTED** [review]

**Comment by FrancoisRegisDegott-eaton** [comment]
deprecated
